src/dos/drive_cache.cpp

```diff
@@ -514,7 +519,7 @@ Bits DOS_Drive_Cache::GetLongName(CFileInfo* curDir, char* shortName) {
 		if (res>0)	low  = mid+1; else
 		if (res<0)	high = mid-1; else
 		{	// Found
-			strcpy(shortName,curDir->fileList[mid]->orgname);
+			safe_strncpy(shortName, curDir->fileList[mid]->orgname, DOS_NAMELENGTH_ASCII);
```

Comment:
`GetLongName` is sometimes called with shortName pointing to buffer of length CROSS_LEN, I need to double check if this is ok.

Reply:
To handle all the variations, I've added a third parameter to `GetLongName(CFileInfo* curDir, char* shortName)` a `const size_t shortName_len`, which is then used in the strncpy.

Reply:
Fixed here: https://github.com/dreamer/dosbox-staging/blob/6c3365e6ae932cbb0fe74a4fd4a2e5ee067bc044/src/dos/drive_cache.cpp#L526